src/matchers/lang/cfamily/CBlock.ts

```diff
@@ -1,26 +1,25 @@
-import { InputState } from "../../InputState";
-import { MatchingLogic } from "../../Matchers";
-import { MatchPrefixResult, matchPrefixSuccess } from "../../MatchPrefixResult";
-import { TerminalPatternMatch } from "../../PatternMatch";
-import { Concat } from "../Concat";
-
-import { inputStateFromString } from "../../internal/InputStateFactory";
-
-import { JavaContentStateMachine } from "./JavaContentStateMachine";
+import { InputState } from "../../../InputState";
+import { inputStateFromString } from "../../../internal/InputStateFactory";
+import { MatchingLogic } from "../../../Matchers";
+import { MatchPrefixResult, matchPrefixSuccess } from "../../../MatchPrefixResult";
+import { TerminalPatternMatch } from "../../../PatternMatch";
+import { Concat } from "../../Concat";
+import { LangStateMachine } from "../LangStateMachine";
 
 /**
- * The rest of a Java block, going to a matching depth of +1 curlies or braces.
+ * The rest of a C family block, going to a matching depth of +1 curlies or braces.
  * Does not read final curly
  */
-class JavaBody implements MatchingLogic {
+export class CBlock implements MatchingLogic {
 
-    public $id: "Java.BlockBody";
+    public $id: "C.BlockBody";
 
     private push: string;
 
     private pop: string;
 
-    constructor(private kind: "block" | "parens", private inner?: MatchingLogic) {
+    constructor(private stateMachineFactory: () => LangStateMachine,
+                private kind: "block" | "parens", private inner?: MatchingLogic) {
```

Comment:
TypeScript enums are a bit weird, so I don't think they're necessarily better. It might look a bit nicer, I guess.